File: apps/net/Ch7_Db/Data/DataLoader.cs
using System.Diagnostics;
using System.Linq;

namespace Ch7_Db.Data
{
    public static class DataLoader
    {
        public static void LoadIfNeeded()
        {
            var db = new AppDbContext();
            db.Database.EnsureCreated();

            using (db)
            {
                Debug.WriteLine(db.Database.CanConnect());
                if (db.Guitars.Any())
                {
                    Debug.WriteLine("Skipping data load, already has guitars loaded.");
                    return;
                }

                Debug.WriteLine("No data found. Loading new guitars...");

                var guitars =
                new[] {
                    new Guitar("AX Black", 499, "/img/guitars/ax-black.jpg", "electric"),
                    new Guitar("Jet Black Electric", 599, "/img/guitars/jet-black-electric.jpg",  "electric"),
                    new Guitar("Weezer Classic", 1499, "/img/guitars/weezer-classic.jpg",  "electric"),
                    new Guitar("Acoustic Black", 1299, "/img/guitars/black-acoustic.jpg",  "acoustic"),
                    new Guitar("Mellow Yellow", 799, "/img/guitars/mellow-yellow.jpg",  "electric"),
                    new Guitar("White Vibes", 699, "/img/guitars/white-vibes.jpg",  "electric"),
                    new Guitar("Brush Riffs", 599, "/img/guitars/brushed-black-electric.jpg",  "electric"),
                    new Guitar("Nature''s Song", 799, "/img/guitars/natures-song.jpg", "electric"),
                    new Guitar("Electric Wood Grain", 399, "/img/guitars/woodgrain-electric.jpg", "electric"),
                };

                foreach (var g in guitars)
                {
                    db.Guitars.Add(g);
                }
                db.SaveChanges();
            }

        }
    }
}

File: apps/py/ch08_db/guitary/data/data_loader.py
from guitary.data.guitar import Guitar
from guitary.data import session_factory


def load_guitars_if_empty():
    with session_factory.create_session() as ctx:
        count = ctx.session.query(Guitar).count()
        if count > 0:
            print(f'Not adding new data, there are {count} guitars already.')
            return

        guitars = [
            Guitar(name='AX Black', price=499, img='/static/img/guitars/ax-black.jpg', style='electric'),
            Guitar(
                name='Jet Black Electric', price=599, img='/static/img/guitars/jet-black-electric.jpg', style='electric'
            ),
            Guitar(name='Weezer Classic', price=1499, img='/static/img/guitars/weezer-classic.jpg', style='electric'),
            Guitar(name='Acoustic Black', price=1298, img='/static/img/guitars/black-acoustic.jpg', style='acoustic'),
            Guitar(name='Mellow Yellow', price=799, img='/static/img/guitars/mellow-yellow.jpg', style='electric'),
            Guitar(name='White Vibes', price=699, img='/static/img/guitars/white-vibes.jpg', style='electric'),
            Guitar(
                name='Brush Riffs', price=599, img='/static/img/guitars/brushed-black-electric.jpg', style='electric'
            ),
            Guitar(name="Nature's Song", price=799, img='/static/img/guitars/natures-song.jpg', style='electric'),
            Guitar(
                name='Electric Wood Grain',
                price=399,
                img='/static/img/guitars/woodgrain-electric.jpg',
                style='electric',
            ),
        ]

        # Do work here...
        for guitar in guitars:
            ctx.session.add(guitar)

        ctx.session.commit()
